src/test/java/teammates/test/cases/webapi/GetFeedbackQuestionRecipientsActionTest.java

```diff
@@ -23,12 +52,190 @@ protected String getRequestMethod() {
+        assertEquals("Team 1.2", teamRecipients.getRecipients().get(0).getIdentifier());
+
+        ______TS("Test typical recipient type: Own team member");
+        loginAsStudent(student1InCourse1.googleId);
+        String[] memberQuestionParams = generateParameters(secondSessionInCourse1, 2,
+                Intent.STUDENT_SUBMISSION, "", "", "");
+        FeedbackQuestionRecipientsData memberRecipients = getRecipients(memberQuestionParams);
+        assertEquals(3, memberRecipients.getRecipients().size());
+        assertEquals("student2InCourse1@gmail.tmt", memberRecipients.getRecipients().get(0).getIdentifier());
+        assertEquals("student3InCourse1@gmail.tmt", memberRecipients.getRecipients().get(1).getIdentifier());
+        assertEquals("student4InCourse1@gmail.tmt", memberRecipients.getRecipients().get(2).getIdentifier());
+
+
+        ______TS("Test typical recipient type: None");
+        loginAsInstructor(instructor1OfCourse1.googleId);
+        String[] noneQuestionParams = generateParameters(firstSessionInCourse1, 3,
+                Intent.INSTRUCTOR_SUBMISSION, "", "", "");
+        FeedbackQuestionRecipientsData noneRecipients = getRecipients(noneQuestionParams);
+        assertEquals(1, noneRecipients.getRecipients().size());
+        assertEquals(Const.GENERAL_QUESTION, noneRecipients.getRecipients().get(0).getIdentifier());
+
     }
 
     @Test
     @Override
     protected void testAccessControl() throws Exception {
-        // TODO
+
+        ______TS("Student intends to access instructor's question, should not be accessible");
+        String[] studentAccessInstructorQuestionParams = generateParameters(firstSessionInCourse1, 3,
+                Intent.STUDENT_SUBMISSION, "", "", "");
+        verifyCannotAccess(studentAccessInstructorQuestionParams);
+
+        ______TS("Instructor intends to access student's question, should not be accessible");
+        String[] instructorAccessStudentQuestionParams = generateParameters(firstSessionInCourse1, 2,
+                Intent.INSTRUCTOR_SUBMISSION, "", "", "");
+        verifyCannotAccess(instructorAccessStudentQuestionParams);
+
+        ______TS("Instructor access instructor's question, should be accessible");
+        String[] instructorSubmissionParams = generateParameters(firstSessionInCourse1, 3,
+                Intent.INSTRUCTOR_SUBMISSION, "", "", "");
+        verifyOnlyInstructorsOfTheSameCourseCanAccess(instructorSubmissionParams);
+        verifyInaccessibleWithoutModifyInstructorPrivilege(instructorSubmissionParams);
+
+        ______TS("Student access student's question, should be accessible");
+        String[] studentSubmissionParams = generateParameters(firstSessionInCourse1, 2,
+                Intent.STUDENT_SUBMISSION, "", "", "");
+        verifyAccessibleForStudentsOfTheSameCourse(studentSubmissionParams);
+
+        ______TS("Student in the same course without logging in, should be accessible");
```

Comment:
`Student in the same course without logging in` means -> `Unregistered student access with correct regKey`